### PR TITLE
Simplify `types.Array` `Append` signature

### PR DESCRIPTION
**Pull request by w84thesun**
# Description

In this PR I remove `error` from the `Append` result parameters as it's always nil.

## Readiness checklist

* [ ] I added unit tests for new functionality or bug fixes.
* [ ] I added integration tests for new functionality or bug fixes.
* [ ] I added compatibility tests for new functionality or bug fixes.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/common/distinct.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ func FilterDistinctValues(docs []*types.Document, key string) (*types.Array, err
 				}
 
 				if !distinct.Contains(el) {
-					err := distinct.Append(el)
+					distinct.Append(el)
 					if err != nil {
```
Comment (AlekSi) [inline internal/handlers/common/distinct.go:119]: I guess that `if` should go away too

**Review by AlekSi — COMMENTED** [review]
TBH, I'm a bit skeptical about that PR. If we ever decide to return some errors from `Append` (for example, validation errors) it will be too easier to forget to add error checking to all invocations back

File: internal/handlers/common/distinct.go (hunk 1 above)
Comment (rumyantseva) [reply internal/handlers/common/distinct.go:119]: Yeah, this `if` is not needed.

(Someone should teach me how to do `if err := distinct.Append(el); err != nil {` next time 😅)

File: internal/handlers/common/distinct.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func FilterDistinctValues(docs []*types.Document, key string) (*types.Array, err
 
 		default:
 			if !distinct.Contains(v) {
-				err := distinct.Append(v)
+				distinct.Append(v)
 				if err != nil {
```
Comment (rumyantseva) [inline internal/handlers/common/distinct.go:128]: Another `err` that is not needed

**Review by rumyantseva — CHANGES_REQUESTED** [review]
A couple of errors doesn't need to be checked anymore, but I like the idea of this improvement!

**Comment by w84thesun** [comment]
> TBH, I'm a bit skeptical about that PR. If we ever decide to return some errors from `Append` (for example, validation errors) it will be too easier to forget to add error checking to all invocations back

I believe that linters could warn about that and it's better to keep that function simple as it is now.

**Review by chilagrow — APPROVED** [review]
Thanks for the simplification.

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**